GMTSAR submission

johndesanto opened the pull request:
Submitting a Portfile to allow users to download the software GMTSAR for satellite radar processing. This software is maintained by Dr. David Sandwell and Dr. Xiaohua Xu at the Scripps Institution of Oceanography. This portfile has passed port lint and been tested successfully on multiple machines with different operating systems.

-John DeSanto

#### Description

<!-- Note: it is best make pull requests from a branch rather than from master -->

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Generate version information with this command in shell:
    echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion)"; echo "Xcode $(xcodebuild -version | awk '{print $NF}' | tr '\n' ' ')"
-->
macOS 10.x
Xcode 8.x

###### Verification <!-- (delete not applicable items) -->
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?
<!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

reneeotten commented:
thanks for the submission! I am not familiar with this software, but looking at the log and the Portfile I noticed two things that strike me as a little odd:

1) it uses both "clang" and "gcc" to compile the software, I would imagine that one should pick one of them and compile everything with the same compiler. If you check the [log](https://paste.macports.org/7a08fd9342c5)  you'll see that it first uses "/usr/bin/clang" and then  it switches to ```gcc``` after  ```(cd epr_api-2.3/src/; make)```.

2) the use of ```configure.args      --with-orbits-dir=/usr/local/orbits``` is a little strange. A quick look on the website suggests that one needs to download ```orbits``` and you presumably did that and put it yourself in ```/usr/local```; so a user will need to do this as well before installing ```gmtsar``` with MacPorts for it to work. If ```orbits``` is required then you should probably take care of downloading/ installing it within the Portfile. I don't know off hand how to do this, but imagine it's described in the guide and/or there are probably other Portfiles that can be used as an example.

Finally, after that's all settled you should squash the commits into one and follow the guidelines for commit messages, so something like ```gmtsar: new port, version 5.6```.

cjones051073 commented:
> thanks for the submission! I am not familiar with this software, but looking at the log and the Portfile I noticed two things that strike me as a little odd:
> 
> 1. it uses both "clang" and "gcc" to compile the software, I would imagine that one should pick one of them and compile everything with the same compiler. If you check the [log](https://paste.macports.org/7a08fd9342c5)  you'll see that it first uses "/usr/bin/clang" and then  it switches to `gcc` after  `(cd epr_api-2.3/src/; make)`.

Yes, that is odd. Does the build/configure not respect the usual build envs (CC, CXX) which MacPorts should be setting ?

> 2. the use of `configure.args      --with-orbits-dir=/usr/local/orbits` is a little strange. A quick look on the website suggests that one needs to download `orbits` and you presumably did that and put it yourself in `/usr/local`; so a user will need to do this as well before installing `gmtsar` with MacPorts for it to work. If `orbits` is required then you should probably take care of downloading/ installing it within the Portfile. I don't know off hand how to do this, but imagine it's described in the guide and/or there are probably other Portfiles that can be used as an example.

its indeed not acceptable to require a user to manually download an external dependency, and then for a port to specifically refer to a location under `/usr/local`. So this has to change. either as above, so update the port to download the required dependency, or perhaps better make a second port for orbits which does this, and installs it into the macports prefix, and then simply make this port depend on it.

> 
> Finally, after that's all settled you should squash the commits into one and follow the guidelines for commit messages, so something like `gmtsar: new port, version 5.6`.

Yes, please squash all commits into a single one.

reneeotten commented:
oh, I see now that this ```ORBITS.tar``` file is about 5.8 GB of data.... that complicates life ;)

cjones051073 commented:
Hmmm. Maybe ask on the devel list the best way to handle this....

cjones051073 commented:
I noticed the `ORBITS.tar` file was not compressed, which struck me as a little unusual, so I got curious and downloaded it. Turns out just compressing it with `bzip2` (which macports uses for its distribution) the size goes down to 1.8G, which is still large but a bit more manageable.... 

Quite why upstream does not compress these files themselves for distribution is beyond me, but that's there choice I guess...

johndesanto commented:
Hi Renee, Chris,

Thank you for touching base with regards to getting GMTSAR integrated with macports. I’m also cc’ing Dave Sandwell and Xiaohua Xu since they know the most about this software and may be able to add some insight. To address your concerns:

1) I’m not sure why exactly the epr_api scripts are delinquent, but suspect we would prefer the entire software to be compiled with gcc. Would the portfile command to set this be configure.cc <http://configure.cc/>? Also, if we were to make this change, would we have to add gcc as a port dependency to ensure the compiler is on the end users machine?

2) The orbits directory contains information on the orbital parameters over time of the ERS and ENVISAT satellites. It is actually not required for the software to compile, but it is required to process data collected by those satellites. This is mostly data files and wouldn’t make much sense to compile and make, so I’m not entirely sure what the appropriate way to put together a port file for these files would be.

3) I confess to being new to github and opened the account specifically to make this submission. With that said, how can I collapse the commits as you are asking?

Thanks,
-John DeSanto

> TOn Mar 23, 2019, at 2:21 PM, Chris Jones <notifications@github.com> wrote:
> 
> I noticed the ORBITS.tar file was not compressed, which struck me as a little unusual, so I got curious and downloaded it. Turns out just compressing it with bzip2 (which macports uses for its distribution) the size goes down to 1.8G, which is still large but a bit more manageable....
> 
> Quite why upstream does not compress these files themselves for distribution is beyond me, but that's there choice I guess...
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/macports/macports-ports/pull/3904#issuecomment-475905772>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AujFz1jG_tuLrF7sS0_56Zx_zTWOa0zZks5vZprXgaJpZM4cCco6>.
>

cjones051073 commented:
> 1) I’m not sure why exactly the epr_api scripts are delinquent, but suspect we would prefer the entire software to be compiled with gcc. Would the portfile command to set this be configure.cc <http://configure.cc/>? Also, if we were to make this change, would we have to add gcc as a port dependency to ensure the compiler is on the end users machine?

The default compiler on macOS these days is clang, and MacPorts strongly prefers ports to use this compiler if they can. Also note `/usr/bin/gcc` is not the GNU gcc compiler anyway on OSX, but a wrapper around `clang`. So in the end the port is anyway being built fine with clang. Its just a strange configuration issue that part of the build is using `clang` and another `gcc`. Please do not switch to using the macports provided gcc compilers (which really are GNU gcc).

> 2) The orbits directory contains information on the orbital parameters over time of the ERS and ENVISAT satellites. It is actually not required for the software to compile, but it is required to process data collected by those satellites. This is mostly data files and wouldn’t make much sense to compile and make, so I’m not entirely sure what the appropriate way to put together a port file for these files would be. 

Ports can do way more than just compiler code ;) We have plenty of examples of ports than simply download and extract data files, so this would not be that out of the ordinary to have something do that here.

That said, does the application *need* to be told, at configure time, where the data will eventually be kept on disc ? Or can this be configured by the user themselves when the run the application. If the later, then another option would be to remove the configuration option

```
configure.args      --with-orbits-dir=/usr/local/orbits
```

and leave it to the user to configure themselves later on.

If the port has to be given the data path at configure time, then I think a port for the ORBITS file that just unpacks the tarball is probably the way to go. If you want to go this way, I can provide examples to work from.

> 3) I confess to being new to github and opened the account specifically to make this submission. With that said, how can I collapse the commits as you are asking? 

There are a number of ways you can do this, but the following is what I normally end up doing.

a) First, I will assume you have checked out the branch here, and the last two commits are the two you want to squash (you can check this with `git log`). Then run

```
git reset --soft HEAD~2
```

what that does is reset the last two commits, back into the staging area. If you run `git log` again after you will see the two commits are gone, and `git status` should show the changes are back staged, ready to be committed again. At this point you can either just re-commit, or if you need to make more changes (for above comments) make those and `git add` them when ready.

Finally, when all your changes are ready just commit them again, following the macPorts guidelines. 

when done, you will need to force push the changes back to the branch in GitHub. force is needed, as you have changed history in the branch.

```
git push --force .....
```

Thanks, -John DeSanto

cjones051073 commented:
One other little detail. The port uses `hdf5` and `tiff` so please add explicit dependencies on these ports to depends_lib

```
depends_lib   port:gmt5 port:hdf5 port:tiff
```

currently these are indirectly depended on, via `gmt5`. The fact the port directly uses them means it should also explicitly depend on them.

johndesanto commented:
Hi Chris,

A quick update. We are currently testing whether we can compile the epr-api scripts using compilers other than gcc without breaking anything. In the meantime, could you please send us an example portfile to download and unpack a tarball file so that we may implement this for the orbits directory?

We’ll address the other concerns you brought up about compressing the github commits and adding hdf5 and tiff as dependencies as we prepare to resubmit.

Thanks,
-John DeSanto

> On Mar 24, 2019, at 4:06 AM, Chris Jones <notifications@github.com> wrote:
> 
> One other little detail. The port uses hdf5 and tiff so please add explicit dependencies on these ports to depends_lib
> 
> depends_lib   port:gmt5 port:hdf5 port:tiff
> currently these are indirectly depended on, via gmt5. The fact the port directly uses them means it should also explicitly depend on them.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/macports/macports-ports/pull/3904#issuecomment-475948490>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AujFz0kYnF0-6dDEbUytjtZO5WS9oJViks5vZ1wtgaJpZM4cCco6>.
>

reneeotten commented:
@johndesanto thanks for the update! While you're looking at it anyway, please also pay attention to some other failures in the log; an example is shown below but it happens more than once if I'm not mistaken:

```
make[2]: Entering directory `/opt/local/var/macports/build/_Users_travis_build_macports_macports-ports_science_gmtsar/gmtsar/work/GMTSAR_V5.6/preproc/ALOS_preproc/ALOS_mosaic_ss'
/usr/bin/clang -pipe -Os -isysroot/Applications/Xcode-10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -arch x86_64 -O2 -Wall -m64 -fPIC -fno-strict-aliasing -std=c99 -I../include -I../../../gmtsar -I/opt/local/include/gmt  -I/opt/local/include -isysroot/Applications/Xcode-10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk  -c -o ALOS_mosaic_ss.o ALOS_mosaic_ss.c
/usr/bin/clang -pipe -Os -isysroot/Applications/Xcode-10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -arch x86_64 -O2 -Wall -m64 -fPIC -fno-strict-aliasing -std=c99 -I../include -I../../../gmtsar -I/opt/local/include/gmt  -I/opt/local/include -isysroot/Applications/Xcode-10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -L/opt/local/lib -Wl,-headerpad_max_install_names -Wl,-syslibroot,/Applications/Xcode-10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -arch x86_64 -m64 -Wl,-x  -o ALOS_mosaic_ss ALOS_mosaic_ss.o -L../lib -lALOS -lm -L/opt/local/var/macports/build/_Users_travis_build_macports_macports-ports_science_gmtsar/gmtsar/work/GMTSAR_V5.6/gmtsar -lgmtsar -L/opt/local/lib -lgmt -I../include -I../../../gmtsar -I/opt/local/include/gmt
ld: library not found for -lgmtsar
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [ALOS_mosaic_ss] Error 1
```

cjones051073 commented:
@johndesanto Looking at the examples I had in mind, they all where a bit more complex than required here, so in the end I ended up implementing a possible port file myself ;) Attached here. You will need to clean it up a little, add appropriate maintainers (see my previous comment) and commit it yourself to this PR.

[gmtsaw-data.zip](https://github.com/macports/macports-ports/files/3008291/gmtsaw-data.zip)

Note that the port installs the orbits data it `${prefix}/share/gmtsar-data/ORBITS`. So thats what you should use in the main port. Also, of course, make the main port depend on the data one.

johndesanto commented:
Hi Renee,

Based upon the error message you sent us, it looks like a similar problem we had encountered when initially testing gmtsar in macports. The issue we had found was that the epr_api directory would not properly make when built by macports but ran without any issue when made manually. After some doing we think that macports was running the protocols for “make" and "make install" in parallel when they really need to run sequentially for the required libraries to be built before compiling.

Likewise, in the example you sent macports is trying to compile a script before the gmtsar library has been built, so we think this is the same issue. Unfortunately, the fix for the instance we found before submission was to rewrite the makefile to hard code the sequence; this would probably be too cumbersome for the entire software. We don’t have a clear idea of how macports is actually building the software, so we don’t know if there is a more elegant solution. Ideally there would be a switch of some sort that we could flip in the portfile that kept macports from doing this. Perhaps you could offer some insight?

-John DeSanto

> On Mar 25, 2019, at 6:11 PM, Renee Otten <notifications@github.com> wrote:
> 
> @johndesanto <https://github.com/johndesanto> thanks for the update! While you're looking at it anyway, please also pay attention to some other failures in the log; an example is shown below but it happens more than once if I'm not mistaken:
> 
> make[2]: Entering directory `/opt/local/var/macports/build/_Users_travis_build_macports_macports-ports_science_gmtsar/gmtsar/work/GMTSAR_V5.6/preproc/ALOS_preproc/ALOS_mosaic_ss'
> /usr/bin/clang -pipe -Os -isysroot/Applications/Xcode-10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -arch x86_64 -O2 -Wall -m64 -fPIC -fno-strict-aliasing -std=c99 -I../include -I../../../gmtsar -I/opt/local/include/gmt  -I/opt/local/include -isysroot/Applications/Xcode-10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk  -c -o ALOS_mosaic_ss.o ALOS_mosaic_ss.c
> /usr/bin/clang -pipe -Os -isysroot/Applications/Xcode-10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -arch x86_64 -O2 -Wall -m64 -fPIC -fno-strict-aliasing -std=c99 -I../include -I../../../gmtsar -I/opt/local/include/gmt  -I/opt/local/include -isysroot/Applications/Xcode-10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -L/opt/local/lib -Wl,-headerpad_max_install_names -Wl,-syslibroot,/Applications/Xcode-10.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -arch x86_64 -m64 -Wl,-x  -o ALOS_mosaic_ss ALOS_mosaic_ss.o -L../lib -lALOS -lm -L/opt/local/var/macports/build/_Users_travis_build_macports_macports-ports_science_gmtsar/gmtsar/work/GMTSAR_V5.6/gmtsar -lgmtsar -L/opt/local/lib -lgmt -I../include -I../../../gmtsar -I/opt/local/include/gmt
> ld: library not found for -lgmtsar
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> make[2]: *** [ALOS_mosaic_ss] Error 1
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/macports/macports-ports/pull/3904#issuecomment-476435330>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AujFz11h-Iax5xfmQmT6STWmHCwgozixks5vaXOjgaJpZM4cCco6>.
>

cjones051073 commented:
b.t.w. One thing Macports will be doing is by default it will run each build phase with the `-j` option, so will run multiple build targets in parallel (note this is not the same as running `make` and `make install` in parallel). If you think your build system has issues with parallel builds (i.e. with the `-j` option) try disabling parallel builds with

```
use_parallel_build  no
```

reneeotten commented:
> b.t.w. One thing Macports will be doing is by default it will run each build phase with the `-j` option, so will run multiple build targets in parallel (note this is not the same as running `make` and `make install` in parallel). If you think your build system has issues with parallel builds (i.e. with the `-j` option) try disabling parallel builds with
> 
> ```
> use_parallel_build  no
> ```

right, I was just typing the same... this could be the reason (and am pretty sure that's what @johndesanto was referring to). An easy test at least to see if that solves the issues; see [ParallelBuilding](https://trac.macports.org/wiki/howto/ParallelBuilding).

johndesanto commented:
Hi Chris, Renee,

Thanks for the advice. We’ll give the use_parallel_build flag a shot.

-John

> On Mar 26, 2019, at 1:19 PM, Renee Otten <notifications@github.com> wrote:
> 
> b.t.w. One thing Macports will be doing is by default it will run each build phase with the -j option, so will run multiple build targets in parallel (note this is not the same as running make and make install in parallel). If you think your build system has issues with parallel builds (i.e. with the -j option) try disabling parallel builds with
> 
> use_parallel_build  no
> right, I was just typing the same... this could be the reason (and am pretty sure that's what @johndesanto <https://github.com/johndesanto> was referring to). An easy test at least to see if that solves the issues; see ParallelBuilding <https://trac.macports.org/wiki/howto/ParallelBuilding>.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/macports/macports-ports/pull/3904#issuecomment-476832416>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AujFzx4MErIOxMVvx1egOwvuom6IdpSGks5vaoDtgaJpZM4cCco6>.
>

johndesanto commented:
Hi Chris, Renee,

I've re-uploaded port files for gmtsar and the orbits tarball as an optional dependency (gmtsar-data). We have addressed the issues with the one directory force-compiling with gcc, and (hopefully) making protocols build in parallel. We've also set a default variant that doesn't download the ORBITS tarball since that is a big file that takes a long time to install. I've also squashed the commits into what I hope has an acceptable format. Please let me know if there are any more issues that need to be addressed.

-John DeSanto

johndesanto commented:
Ok, so this is embarrassing, but Ive just noticed a typo in the gmtsar portfile where the prot variant that is supposed to be the default is misnamed by one character. This doesn't affect the intended functionality of the portfile but is something I would like to fix even if it is trivial. Is there a way for me to make this change in github without having to recommit?

-John

reneeotten commented:
@johndesanto you can certainly still make changes to the Portfile. You'll have to commit again though, but you can amend the current commit. So, make the required changes and then just to ```git commit -a --amend```. 
Please hold off though for a little bit, there might be a few more things that need to be changed. First, I will ask on the mailinglist what the proper way of taking care of this data file. I think we should make it non-distributable, meaning that a user will download it from the source website instead of the MacPorts mirrors (I don't see the point of storing this amount of data on the mirrors).

reneeotten commented on science/gmtsar/Portfile:
```diff
@@ -0,0 +1,50 @@
+
+variant orbits description {Optional support for ERS and ENVISat satellites} {
+    depends_lib-append      port:gmtsar-data
+    configure.args-append   --with-orbits-dir=/opt/local/share/gmtsar-data/ORBITS
```
please don't hardcode ```/opt/local```, use ```${prefix}``` instead; it's possible to install MacPorts in another location than /opt/local

ryandesign replied:
This should be `${prefix}` instead of `/opt/local`.

ryandesign commented on science/gmtsar-data/Portfile:
```diff
@@ -0,0 +1,46 @@
+
+extract.suffix      .tar
+extract.cmd         tar
+extract.pre_args    -xvf
```
You should replace these `extract` overrides with just `use_tar yes`.

ryandesign commented on science/gmtsar-data/Portfile:
```diff
@@ -0,0 +1,46 @@
+extract.cmd         tar
+extract.pre_args    -xvf
+
+distname            ORBITS
```
Since the `distname` does not contain the version number, you should set `dist_subdir ${name}/${version}`. See https://trac.macports.org/wiki/PortfileRecipes#unversioned-distfiles.

ryandesign commented on science/gmtsar-data/Portfile:
```diff
@@ -0,0 +1,46 @@
+extract.pre_args    -xvf
+
+distname            ORBITS
+distfiles           ${distname}${extract.suffix}
```
You can remove this `distfiles` line, since this is the default value.

ryandesign commented on science/gmtsar-data/Portfile:
```diff
@@ -0,0 +1,46 @@
+
+destroot {
+    set datadir ${prefix}/share/${name}
+    file mkdir "${destroot}${datadir}"
```
Quotes are unnecessary here. Portfiles are Tcl files, not shell scripts. Quoting would be necessary in a shell script to accommodate any possible spaces in the variable values, but Tcl does not have that same problem.

mf2k replied:
Shouldn't `xinstall -d` be used instead?

`xinstall -d ${destroot}${datadir}`

reneeotten commented on science/gmtsar-data/Portfile:
```diff
@@ -0,0 +1,46 @@
+
+destroot {
+    set datadir ${prefix}/share/${name}
+    file mkdir "${destroot}${datadir}"
```
please change to ```xinstall -d ${destroot}${datadir}```

ryandesign commented on science/gmtsar-data/Portfile:
```diff
@@ -0,0 +1,46 @@
+PortSystem          1.0
+
+name                gmtsar-data
+version             1.0
```
Is the data actually tagged with this `1.0` version number, or did you just make it up? If you made it up, then I suggest changing it to a *YYYYMMDD* value, such as the file's modification date, 20150202.

ryandesign commented on science/gmtsar-data/Portfile:
```diff
@@ -0,0 +1,46 @@
+
+maintainers         ucsd.edu:dsandwell \
+                    ucsd.edu:xix016 \
+                    ucsd.edu:jdesanto \
```
Please add a GitHub handle to each one, if available, in the form `{obfuscated:email @githubhandle}`.

ryandesign commented on science/gmtsar-data/Portfile:
```diff
@@ -0,0 +1,46 @@
+                    sha256  35e49d6c51c461566ac7bcaff5ead3243b89957d99f02eca2d969b8cf49396bf \
+                    size    6246106112
+
+extract.dir         ${workpath}/ORBITS
```
Once you're using `use_tar yes`, this `extract.dir` line is probably unnecessary too.

ryandesign commented on science/gmtsar/Portfile:
```diff
@@ -0,0 +1,50 @@
+license             GPL-3
+maintainers         ucsd.edu:dsandwell \
+                    ucsd.edu:xix016 \
+                    ucsd.edu:jdesanto \
```
As above, add GitHub handles if available.

reneeotten commented on science/gmtsar-data/Portfile:
```diff
@@ -0,0 +1,46 @@
+version             1.0
+categories          science
+platforms           darwin
+license             GPL-3
```
please set the ```license``` to ```Restrictive Nomirror```, this will avoid mirroring the distfile and avoid storing archives; it will just always be downloaded from ftp://topex.ucsd.edu/pub/gmtsar/. See the [mailinglist](https://lists.macports.org/pipermail/macports-dev/2019-March/040394.html) for more information.

ryandesign commented on science/gmtsar/Portfile:
```diff
@@ -0,0 +1,50 @@
+                    ucsd.edu:xix016 \
+                    ucsd.edu:jdesanto \
+                    openmaintainer
+description         GMTSAR
```
This `description` is too short; it doesn't really describe anything. What you've written in the `long_description` field is probably the right length for the `description` field; then I'd expand the `long_description` to a sentence or two.

ryandesign commented on science/gmtsar/Portfile:
```diff
@@ -0,0 +1,50 @@
+
+extract.suffix      .tar
+extract.cmd         tar
+extract.pre_args    -xvf
```
As above, use `use_tar yes` instead.

ryandesign commented on science/gmtsar/Portfile:
```diff
@@ -0,0 +1,50 @@
+extract.pre_args    -xvf
+
+distname            GMTSAR_V${version}
+distfiles           ${distname}${extract.suffix}
```
As above, remove this `distfiles` since it is the default.

ryandesign commented on science/gmtsar/Portfile:
```diff
@@ -0,0 +1,50 @@
+
+use_autoconf        yes
+
+use_configure       yes
```
Remove `use_configure yes`; that's the default.

ryandesign commented on science/gmtsar/Portfile:
```diff
@@ -0,0 +1,50 @@
+use_parallel_build  no
+
+variant norbits description {Standard installation for S1A, ALOS, ALOS-2, CSK, TSX, and RS2 satellites} {
+}
```
I'd remove the n(o)orbits variant and the `default_variants` line. A user who wants orbits will enable that variant; a user who doesn't won't. This variant description text could be reworked into the port's `long_description`.

When the user's choice is just to enable a feature or not, that only warrants the creation of a single variant. (Think of it like a GUI checkbox.) But when the user's choice is between two alternatives (such as "x11 or quartz", or "openssl or gnutls"), multiple variants are best, though if the alternatives are mutually exclusive, don't forget to mark them as conflicting with one another. (In this case, think of them like GUI radio buttons.)

johndesanto commented:
Hi All,

I have implemented all of the changes that were recommended, save for removing the "extract.dir" definition in the gmtsar-data portfile. Despite my best efforts, I could not figure out a way for removing this line not to break the installation.

Thanks for all of your advice.
-John

cjones051073 commented:
LGTM now, thanks for all the work with the changes. 
Think we have had enough discussion, so will merge.

reneeotten commented:
@cjones051073 except for the fact that the discussion on the mailinglist regarding the data files was still on-going... And the opinion seems to be not to install this through a port - how to deal with this now?

johndesanto commented:
Hi Chris, Renee,

Glad to hear that the portfile is working. I have spoken with Dave Sandwell, who is one of the lead developers of the software, and we agree that installing the data files is probably too cumbersome through macports, especially since they are not strictly required for the software to install. I can remove the data portfile and variant from the gmtsar portfile and resubmit if that will cleanly resolve this issue.

Apologies for the extra hassle.
-John
